### ISSUE TITLE
Need to login every time app get opens

### ISSUE THREAD

**Issue by mukesh8293** [charlieAndroidDev/Instagram4Android#104]
Is there any way to avoid login always.
any way to save cookies or access token, so that I don't need to log in always?
if yes, please let me know.
Thanks.

**Comment by Vioxini** [comment]
I have a solution but I am unsure of its effectiveness given the amount of times I do have to log in and log out on testings.

You can save Cookie Jar into database and call database and put into Cookie Jar.

**Comment by mukesh8293** [comment]
> I have a solution but I am unsure of its effectiveness given the amount of times I do have to log in and log out on testings.
> 
> You can save Cookie Jar into database and call database and put into Cookie Jar.

>I want to save the Cookies into the database but I am not getting cookies.

**Comment by cellojaan** [comment]
This library could not support to login with cookies. For example, I do not use anymore this library and I done my own library. You should do like this. There is no way to do with this library. Do not search it.

**Comment by Vioxini** [comment]
This library as is cannot support cookie login, it is true. I forgot that I did my own modifications to it.

**Comment by mukesh8293** [comment]
Can I use your library?

**Comment by Vioxini** [comment]
Yes of course. It is a Fork of this one. Anything you need to know give me a touch.

**Comment by stevenkader** [comment]
> Yes of course. It is a Fork of this one. Anything you need to know give me a touch.

Hi, I would like to try your fork, but not sure how to include it in my project.  Any help will be appreciated.  Thanks.

**Comment by stevenkader** [comment]
Hi Vioxini,
Do you have a sample of how to store current login info and then use it next time app opens when calling Instagram4Android.builder( ....)?  Thx

**Comment by simgekeser** [comment]
> Yes of course. It is a Fork of this one. Anything you need to know give me a touch.

how to login for accounts with 2FA auth? Can you give a sample 2FA auth?

**Comment by Vioxini** [comment]
> 
> 
> > Yes of course. It is a Fork of this one. Anything you need to know give me a touch.
> 
> how to login for accounts with 2FA auth? Can you give a sample 2FA auth?

You have to check for 2FA on normal login, then throw a dialog asking for the code, then use ig.login(2factorcode)

:)
